Source/UI/IdentityUI.swift

```diff
@@ -105,6 +105,51 @@ public class IdentityUI {
     // 2. When handling a universal link, the currently presented process (if any) can be retrieved.
     private static var presentedIdentityUI: IdentityUI?
 
+    private static var updatedTermsCoordinator: UpdatedTermsCoordinator?
+
+    /**
+     Present a screen where the user can review and accept updated terms and conditions.
+
+     In order to comply with privacy regulations, you need to make sure that the user accepts any update to terms and conditions that may have been issued since
+     the last visit. For this reason, at the startup of your app, right after having obtained the `IdentityManager.currentUser` and verified the login status,
+     if the user is logged-in you should call `user.agreements.status(:)` and, in case of `false` result (meaning the user has yet to accepted the latest
+     terms), obtain the latest terms by calling `IdentityManager.fetchTerms(:)` and finally call this method to present the updated terms. If the user fails to
+     accept the terms, a logout will automatically be forced.
+
+     - parameter: terms: The terms that the user should accept. They should be obtained by calling `IdentityManager.fetchTerms(:)`.
+     - parameter: user: The user that requires acceptance of new terms. It is important that you pass the same instance of `User` you previously obtained from
+       an `IdentityManager` and stored, otherwise you won't get logout notifications for that user in case the user is logged out for not having accepted the
+       new terms.
+     - parameter: viewController: The view controller to present the screen from.
+     - parameter: configuration: The UI configuration.
+     */
+    public static func presentTerms(_ terms: Terms, for user: User, from viewController: UIViewController, configuration: IdentityUIConfiguration) {
+        guard self.presentedIdentityUI == nil, self.updatedTermsCoordinator == nil else {
+            // Another screen of the Identity UI is already presented.
+            return
```

Comment:
Should we puke some stuff here since this should in theory never happen? precondifitionFailure maybe or just a print statement or something?

Reply:
Very theoretically, the user could log out and start a new login flow while the network calls to fetch the acceptance status and the terms are made, so in that particular case I think it might be possible you end up here with another flow already in place and you just want to ignore this one. But I should probably check and fail if the user is still logged in && another flow is in place, as that should surely never happen.